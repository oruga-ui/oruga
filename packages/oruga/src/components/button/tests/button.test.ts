import { describe, test, expect, afterEach } from "vitest";
import { enableAutoUnmount, mount } from "@vue/test-utils";

import OButton from "@/components/button/Button.vue";

describe("OButton tests", () => {
    enableAutoUnmount(afterEach);

    test("render correctly", () => {
        const wrapper = mount(OButton);
        expect(!!wrapper.vm).toBeTruthy();
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.attributes("data-oruga")).toBe("button");
        expect(wrapper.html()).toMatchSnapshot();
    });
});
