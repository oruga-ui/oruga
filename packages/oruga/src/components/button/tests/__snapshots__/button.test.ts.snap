// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OButton tests > render correctly 1`] = `"<button type="button" class="o-btn" role="button" data-oruga="button"><span class="o-btn__wrapper"><!--v-if--><!--v-if--><!--v-if--></span></button>"`;
